### PR TITLE
Improve avatars

### PR DESCRIPTION
**Pull request by duk3luk3**
* More error handling
* More tests
* Bugfixes
* Handle timestamps

### CODE REVIEW

File: api/error.py [hunk 1]
```diff
@@ -17,6 +17,14 @@ class ErrorCode(Enum):
         code=400,
         title='File exists',
         detail='Avatar file already exists')
+    AVATAR_IN_USE = dict(
+        code=400,
+        title='Avatar in use',
+        detail='Avatar cannot be deleted because it is still in use')
+    AVATAR_INTEGRITY_ERROR = dict(
+        code=400,
```
Comment (Brutus5000) [inline api/error.py]: Code 400 is already in use.

File: api/players.py [hunk 2]
```diff
@@ -27,12 +27,13 @@ def get_active_players():
...
     prefix = prefix.replace('%', '\\%')
-    prefix = prefix.replace('_', '\\%')
+    prefix = prefix.replace('_', '\\_')
     return fetch_data(PlayerSchema(), PLAYER_TABLE, PLAYER_SELECT_EXPRESSIONS, 1000, request, many=True, limit=False, sort='login',
                       where='LOWER(l.login) LIKE %s', args=(prefix.lower() + '%',))
 
 @app.route('/players/<int:player_id>')
-def get_player(player_id):
+@app.route('/players/byname/<string:player_name>')
+def get_player(player_id=None, player_name=None):
```
Comment (micheljung) [inline api/players.py]: If I'd still care about this API, I'd tell you that this route definition isn't JSON-API conform (you'd use filters instead).
Also, using one function with different parameters (and in this case, routes) and then if-else them is a big antipattern. You'd just use two different functions.

File: api/error.py (hunk 1 above)
Comment (micheljung) [reply api/error.py]: Yes, all the error codes are a continuous numbering starting at 100, not HTTP status codes. And neither the error title nor the detail message isn't useful.

File: api/error.py [hunk 3]
```diff
@@ -198,15 +186,35 @@ class ErrorCode(Enum):
...
     )
     AUTHENTICATION_NEEDED = dict(
-        code=401,
+        code=144,
         title='Unauthorized',
         detail='You are not logged in'
     )
     FORBIDDEN = dict(
-        code=403,
+        code=145
```
Comment (Brutus5000) [inline api/error.py]: missing comma -> syntax error

Comment (duk3luk3) [reply api/error.py]: Ah jeez.

**Pull request merged**